air/src/options.rs

```diff
@@ -151,19 +153,26 @@ impl ExecutionOptions {
     // --------------------------------------------------------------------------------------------
 
     /// Creates a new instance of [ExecutionOptions] from the specified parameters.
-    pub fn new(
-        max_cycles: Option<u32>,
-        expected_cycles: u32,
-    ) -> Result<Self, ExecutionOptionsError> {
-        if max_cycles.is_some_and(|max_cycles| max_cycles < expected_cycles) {
-            return Err(ExecutionOptionsError::ExpectedCyclesTooBig(
-                max_cycles.unwrap(),
-                expected_cycles,
-            ));
+    pub fn new(max_cycles: u32, expected_cycles: u32) -> Result<Self, ExecutionOptionsError> {
+        if expected_cycles < MIN_TRACE_LEN as u32 {
+            return Err(ExecutionOptionsError::ExpectedCyclesTooSmall(expected_cycles));
+        }
+        if max_cycles < expected_cycles {
+            return Err(ExecutionOptionsError::ExpectedCyclesTooBig(max_cycles, expected_cycles));
         }
         Ok(ExecutionOptions {
             max_cycles,
             expected_cycles,
         })
     }
```

Comment:
About first point: but if the `expexted_cycles` is not provided at the parsing stage it will be set to the default which is u32::MAX. Why do we need to perform this check again?

Reply:
I think it is cleaner to have the interface of `new()` function be:
```Rust
pub fn new(expected_cycles: u32, max_cycles: Option<u32>) -> Result<Self, ExecutionOptionsError>
```
But we could also have a default at CLI level - I don't think that will complicate the code too much.